split_to_CMYK.jsx

```diff
@@ -1 +1,109 @@
+    levelsDes1.putReference(c2t("null"), levelsRef);
+    levelsDes2.putString(s2t("name"), (props.name) + """ Levels""");
+    levelsDes2.putBoolean(s2t("group"), true);
+    levelsDes3.putEnumerated(s2t("presetKind"), s2t("presetKindType"), s2t("presetKindDefault"));
+    levelsDes2.putObject(s2t("type"), s2t("levels"), levelsDes3);
+    levelsDes1.putObject(s2t("using"), s2t("adjustmentLayer"), levelsDes2);
+    executeAction(s2t("make"), levelsDes1, DialogModes.NO);
+  //add hue sat adj layer - if statement is so Black doesnt get a hue adj layer
+    if(props.name != 'Black') {
+      var addHueSatDes1 = new ActionDescriptor();
+      var addHueSatDes2 = new ActionDescriptor();
+      var addHueSatDes3 = new ActionDescriptor();
+      var addHueSatRef = new ActionReference();
+
+      addHueSatRef.putClass(s2t("adjustmentLayer"));
+      addHueSatDes3.putReference(c2t("null"), addHueSatRef);
+      addHueSatDes1.putString(s2t("name"), (props.name) + """ color adjust""");
+      addHueSatDes1.putBoolean(s2t("group"), true );
+      addHueSatDes2.putEnumerated(s2t("presetKind"), s2t("presetKindType"), s2t("presetKindDefault"));
+      addHueSatDes2.putBoolean(s2t("colorize"), true);
+      addHueSatDes1.putObject(s2t("type"), s2t("hueSaturation"), addHueSatDes2);
+      addHueSatDes3.putObject(s2t("using"), s2t("adjustmentLayer"), addHueSatDes1);
+      executeAction(s2t("make"), addHueSatDes3, DialogModes.NO);
+      //sets hue, sat, lightness
+      var setHueSatDes1 = new ActionDescriptor();
+      var setHueSatDes2 = new ActionDescriptor();
+      var setHueSatDes3 = new ActionDescriptor();
+      var setHueSatList = new ActionList();
+      var setHueSatRef = new ActionReference();
+
+      setHueSatRef.putEnumerated(s2t("adjustmentLayer"), s2t("ordinal"), s2t("targetEnum"));
+      setHueSatDes1.putReference(c2t("null"), setHueSatRef);
+      setHueSatDes2.putEnumerated(s2t("presetKind"), s2t("presetKindType"), s2t("presetKindCustom"));
+      setHueSatDes3.putEnumerated(s2t("channel"), s2t("channel"), s2t("composite"));
+      setHueSatDes3.putInteger(s2t("hue"), props.h);
+      setHueSatDes3.putInteger(c2t("Strt"), props.s);
+      setHueSatDes3.putInteger(s2t("lightness"), props.v);
+      setHueSatList.putObject(s2t("hueSatAdjustmentV2"), setHueSatDes3);
+      setHueSatDes2.putList(s2t("adjustment"), setHueSatList);
+      setHueSatDes1.putObject(s2t("to"), s2t("hueSaturation"), setHueSatDes2);
+      executeAction(s2t("set"), setHueSatDes1, DialogModes.NO);
+    }; else {return};
+  };
+};
+function runScript() {
+  var CYAN_LAYER = new ChannelLayer('Cyan', BlendMode.NORMAL, 174, 50, 20);
+  var MAGENTA_LAYER = new ChannelLayer('Magenta', BlendMode.MULTIPLY, 300, 50, 20);
+  var YELLOW_LAYER = new ChannelLayer('Yellow', BlendMode.MULTIPLY, 50, 50, 20);
+  var BLACK_LAYER = new ChannelLayer('Black', BlendMode.MULTIPLY, 0, 0, 0);
+  var layers = [CYAN_LAYER, MAGENTA_LAYER, YELLOW_LAYER, BLACK_LAYER];
```

Comment:
And, actually now that I'm looking over lines 92-96 again you _could_ express this more simply like:
```
var layers = [ new ChannelLayer('Cyan', BlendMode.NORMAL, 174, 50, 20),
                 new ChannelLayer('Magenta', BlendMode.MULTIPLY, 300, 50, 20),
                 new ChannelLayer('Yellow', BlendMode.MULTIPLY, 50, 50, 20),
                 new ChannelLayer('Black', BlendMode.MULTIPLY, 0, 0, 0) ];
```

Since you're not actually referencing each layer name anywhere, you can declare each anonymously at the same time as you declare your layers Array.  Saves you some var clutter, eliminates a line of code, and IMO clarifies the intent a little better.  Up to you if you want to change this though since it's functionally fine either way.